paparazzi/src/test/java/app/cash/paparazzi/HtmlReportWriterTest.kt

```diff
@@ -113,48 +114,156 @@ class HtmlReportWriterTest {
 
   @Test
   fun alwaysOverwriteOnRecord() {
-    // set record mode
-    System.setProperty("paparazzi.test.record", "true")
+    try {
+      // set record mode
+      System.setProperty("paparazzi.test.record", "true")
 
-    val htmlReportWriter = HtmlReportWriter("record_run", reportRoot.root, snapshotRoot.root)
+      val htmlReportWriter = HtmlReportWriter("record_run", reportRoot.root, snapshotRoot.root)
+      htmlReportWriter.use {
+        val now = Instant.parse("2021-02-23T10:27:43Z")
+        val snapshot = Snapshot(
+          name = "test",
+          testName = TestName("app.cash.paparazzi", "HomeView", "testSettings"),
+          timestamp = now.toDate()
+        )
+        val golden =
+          File("${snapshotRoot.root}/images/app.cash.paparazzi_HomeView_testSettings_test.png")
+
+        // precondition
+        assertThat(golden).doesNotExist()
+
+        // take 1
+        val frameHandler1 = htmlReportWriter.newFrameHandler(
+          snapshot = snapshot,
+          frameCount = 1,
+          fps = -1
+        )
+        frameHandler1.use { frameHandler1.handle(anyImage) }
+        assertThat(golden).exists()
+        val timeFirstWrite = golden.lastModifiedTime()
+
+        // I know....but guarantees writes won't happen in same tick
+        Thread.sleep(100)
+
+        // take 2
+        val frameHandler2 = htmlReportWriter.newFrameHandler(
+          snapshot = snapshot.copy(timestamp = now.plusSeconds(1).toDate()),
+          frameCount = 1,
+          fps = -1
+        )
+        frameHandler2.use { frameHandler2.handle(anyImage) }
+        assertThat(golden).exists()
+        val timeOverwrite = golden.lastModifiedTime()
+
+        // should always overwrite
+        assertThat(timeOverwrite).isGreaterThan(timeFirstWrite)
+      }
+    } finally {
+      // reset record mode
+      System.setProperty("paparazzi.test.record", "false")
+    }
+  }
+
+  @Test
+  fun happyPathVideos() {
+    val htmlReportWriter = HtmlReportWriter("run_one", reportRoot.root)
     htmlReportWriter.use {
-      val now = Instant.parse("2021-02-23T10:27:43Z")
-      val snapshot = Snapshot(
-        name = "test",
-        testName = TestName("app.cash.paparazzi", "HomeView", "testSettings"),
-        timestamp = now.toDate()
+      val frameHandler = htmlReportWriter.newFrameHandler(
+        Snapshot(
+          name = "loading",
+          testName = TestName("app.cash.paparazzi", "CelebrityTest", "testSettings"),
+          timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+          tags = listOf("redesign")
+        ),
+        2,
+        1
```

Comment:
let's use named arguments here to make the test more readable (same in the predecessor, `happyPathImages`)